lib/spack/spack/mirror.py

```diff
@@ -422,27 +479,25 @@ def create(path, specs, skip_unstable_versions=False):
     it creates specs for those versions.  If the version satisfies any spec
     in the specs list, it is downloaded and added to the mirror.
     """
-    parsed = url_util.parse(path)
+    parsed = url_util.parse(url)
     mirror_root = url_util.local_file_path(parsed)
-    if not mirror_root:
-        raise spack.error.SpackError(
-            'MirrorCaches only work with file:// URLs')
 
     # automatically spec-ify anything in the specs array.
     specs = [
         s if isinstance(s, spack.spec.Spec) else spack.spec.Spec(s)
         for s in specs]
 
-    # Get the absolute path of the root before we start jumping around.
-    if not os.path.isdir(mirror_root):
-        try:
-            mkdirp(mirror_root)
-        except OSError as e:
-            raise MirrorError(
-                "Cannot create directory '%s':" % mirror_root, str(e))
+    if mirror_root:
+        # Get the absolute path of the root before we start jumping around.
+        if not os.path.isdir(mirror_root):
```

Comment:
Minor change, but I would rename `mirror_root` to `local_mirror_root` and have a single conditional here:
```python
if local_mirror_root and not os.path.isdir(local_mirror_root):
    ...
```